test/e2e/testkit/otlp/metrics/metrics.go

```diff
@@ -183,6 +183,46 @@ func makeUnique(slice []string) []string {
 	return uniqueList
 }
 
+func AllDataPointsHaveAttributes(m pmetric.Metric, expectedAttrKeys ...string) bool {
+	var attrsPerDataPoint []pcommon.Map
+
+	switch m.Type() {
+	case pmetric.MetricTypeSum:
+		for i := 0; i < m.Sum().DataPoints().Len(); i++ {
+			attrsPerDataPoint = append(attrsPerDataPoint, m.Sum().DataPoints().At(i).Attributes())
```

Comment:
Do `m.Sum().DataPoints()`, `m.Gauge().DataPoints()`, etc. return the same instance? If so, maybe we can simplify these 4 loops into just one?